src/main/java/nl/tudelft/ewi/devhub/server/database/entities/Group.java

```diff
@@ -24,20 +25,27 @@
 @Data
 @Entity
 @Table(name = "groups")
-@EqualsAndHashCode(of = { "groupId" })
-@ToString(of = { "groupId" })
-public class Group implements Serializable {
+@IdClass(Group.GroupId.class)
+@ToString(of = { "course", "groupNumber" })
+@EqualsAndHashCode(of = { "course", "groupNumber" })
+public class Group {
 
-	@Id
-	@Column(name = "id")
-	@GeneratedValue(strategy = GenerationType.IDENTITY)
-	private long groupId;
+	@Data
+	@EqualsAndHashCode
+	public static class GroupId implements Serializable {
+
+		private Course course;
+		private Long groupNumber;
+
+	}
 
+	@Id
```

Comment:
`@Id` on `@ManyToOne` columns is not allowed. We should use [`@MapsId`](http://docs.oracle.com/javaee/6/api/javax/persistence/MapsId.html) instead.